lisa/sut_orchestrator/baremetal/build.py

```diff
@@ -79,23 +105,54 @@ def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
                         )
                 self._log.debug(f"copy file {file} to {share_name}\\{file_name}")
 
-    def _find_matched_files(
-        self, sources_path: List[Path], files_map: List[FileSchema]
-    ) -> Dict[str, FileSchema]:
-        all_files = []
-        match_files: Dict[str, FileSchema] = {}
-        for source_path in sources_path:
-            for root, _, files in os.walk(source_path):
-                for file in files:
-                    all_files.append(os.path.join(root, file))
-
-            for file_map in files_map:
-                file_path = rf"{source_path}\{file_map.source}".replace("\\", "\\\\")
-                pattern = re.compile(
-                    file_path,
-                    re.I | re.M,
+
+class TftpBuild(Build):
+    def __init__(self, runbook: TftpBuildSchema) -> None:
+        super().__init__(runbook)
+        self.pxe_runbook: TftpBuildSchema = self.runbook
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "tftp"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return TftpBuildSchema
+
+    def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
+        assert self.pxe_runbook.connection, "The build server is not specified"
+
+        try:
+            build_server = quick_connect(
+                self.pxe_runbook.connection,
+                logger_name="build_server",
+            )
+        except Exception as ex:
+            self._log.debug(
+                f"failed to connect {self.pxe_runbook.connection}",
+                exc_info=ex,
+            )
+
+        ls = build_server.tools[Ls]
+        rc = build_server.tools[RemoteCopy]
+        for file, file_map in _find_matched_files(
+            sources_path,
+            files_map,
+        ).items():
+            if file_map.destination:
+                if ls.is_file(PurePath(file_map.destination)):
+                    file_destination = os.path.dirname(PurePath(file_map.destination))
+                else:
+                    file_destination = file_map.destination
+            else:
+                file_destination = os.path.dirname(PurePath(file))
+
+            try:
+                rc.copy_to_remote(PurePath(file), PurePath(file_destination))
+            except Exception as ex:
```

Comment:
Be careufly when catching and eating exceptions. File copy is a critical step, if the file is not copied, code should fail here. So users know something wrong, and fix it. If the exception is ate here, the error is hard to find in next steps.

Reply:
Done

Reply:
`log.exception` doesn't raise an exception. It needs to call like `raise LisaException(...` to rewrite the exception message and raise it.

Reply:
The change is done, thanks